lib/rubocop/ast/traversal.rb

```diff
@@ -18,15 +19,21 @@ def walk(node)
                              rational str sym regopt self lvar
                              ivar cvar gvar nth_ref back_ref cbase
                              arg restarg blockarg shadowarg
-                             kwrestarg zsuper lambda redo retry].freeze
+                             kwrestarg zsuper lambda redo retry
+                             forward_args forwarded_args
+                             match_var match_nil_pattern].freeze
       ONE_CHILD_NODE    = %i[splat kwsplat block_pass not break next
                              preexe postexe match_current_line defined?
-                             arg_expr].freeze
+                             arg_expr pin match_rest if_guard unless_guard
+                             match_with_trailing_comma].freeze
       MANY_CHILD_NODES  = %i[dstr dsym xstr regexp array hash pair
                              mlhs masgn or_asgn and_asgn
                              undef alias args super yield or and
                              while_post until_post iflipflop eflipflop
-                             match_with_lvasgn begin kwbegin return].freeze
+                             match_with_lvasgn begin kwbegin return
+                             in_match case_match in_pattern match_alt
+                             match_as array_pattern array_pattern_with_tail
+                             hash_pattern const_pattern].freeze
```

Comment:
I just noticed that the traversal methods generated via `MANY_CHILD_NODES` assume that every child node exists (i.e. is not nil). That is not true for `in_pattern` nodes, for example. So we need to review those types more carefully.